lib/java/desktop-environment/src/main/java/org/enso/desktopenvironment/LinuxTrashBin.java

```diff
@@ -0,0 +1,220 @@
+    private static final int MAX_ATTEMPTS = Byte.MAX_VALUE;
+    private static final String TRASHINFO_EXTENSION = ".trashinfo";
+
+    /**
+     * Create the .trashinfo file containing the deleted file metadata.
+     *
+     * @param trashInfo the path to the trashinfo directory.
+     * @param toDelete the path to the file that should be deleted.
+     * @return the trashinfo metadata file.
+     * @throws IOException if the file creation was unsuccessful.
+     */
+    public static TrashInfo create(Path trashInfo, Path toDelete) throws IOException {
+      var builder = new StringBuilder();
+      builder.append("[Trash Info]");
+      builder.append(System.lineSeparator());
+      builder.append("Path=");
+      builder.append(toDelete.toAbsolutePath());
+      builder.append(System.lineSeparator());
+      builder.append("DeletionDate=");
+      builder.append(
+          LocalDateTime.now()
+              .truncatedTo(ChronoUnit.SECONDS)
+              .format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
+      builder.append(System.lineSeparator());
+
+      return createTrashInfo(trashInfo, toDelete, builder, "", 0);
+    }
+
+    /**
+     * Create the .trashinfo file containing the deleted file metadata.
+     *
+     * <p>In case of a name clash, when the trash already contains the file with the same name, the
+     * trashinfo file is created with a random suffix to resolve the conflict.
+     *
+     * <p>The file creation is atomic to so that if two processes try trash files with the same
+     * filename this will result in two different trash files.
+     *
+     * @param trashInfo the path to the trashinfo directory.
+     * @param toDelete the path to the file that should be deleted.
+     * @param contents the trashinfo file contents.
+     * @param suffix the trashinfo suffix to resolve the file name conflicts.
+     * @param attempts the number of attempts to resolve the name clash.
+     * @return the trashinfo metadata file.
+     * @throws IOException if the file creation was unsuccessful.
+     */
+    private static TrashInfo createTrashInfo(
+        Path trashInfo, Path toDelete, CharSequence contents, String suffix, int attempts)
+        throws IOException {
+      if (attempts > MAX_ATTEMPTS) {
+        throw new IOException("Failed to create trashinfo file. Max attempts reached.");
+      }
+
+      try {
+        var fileName = toDelete.getFileName().toString() + suffix;
+        var path =
+            Files.writeString(
+                trashInfo.resolve(fileName + TRASHINFO_EXTENSION),
+                contents,
+                StandardCharsets.UTF_8,
+                StandardOpenOption.CREATE_NEW,
+                StandardOpenOption.WRITE);
+
+        return new TrashInfo(path, fileName);
+      } catch (FileAlreadyExistsException e) {
+        return createTrashInfo(
+            trashInfo,
+            toDelete,
+            contents,
+            RandomUtils.alphanumericString(SUFFIX_SIZE),
```

Comment:
It is quite common in Enso to use `UUID` - which is usually created randomly based on various conditions including time seed. Is not it easier to remove `RandomUtils` and just ask for `UUID`?